### PR TITLE
Py-scitools-iris: python library for analysing and visualising Earth science data

### PR DESCRIPTION
**Pull request by mjwoods**
According to the documentation at https://scitools.org.uk/iris/docs/latest/:

> Iris implements a data model based on the CF conventions giving you a powerful, format-agnostic interface for working with your data. It excels when working with multi-dimensional Earth Science data, where tabular representations become unwieldy and inefficient.

Iris can read, write and visualise gridded data in netcdf, grib and UM formats. It is especially useful when working with the Unified Model, which is developed by the UK Met Office and numerous collaborators for weather and climate forecasts.

### CODE REVIEW

**Comment by citibeth** [comment]
> According to the documentation at https://scitools.org.uk/iris/docs/latest/:
> 
> > Iris implements a data model based on the CF conventions giving you a powerful, format-agnostic interface for working with your data. It excels when working with multi-dimensional Earth Science data, where tabular representations become unwieldy and inefficient.
> 
> Iris can read, write and visualise gridded data in netcdf, grib and UM formats. It is especially useful when working with the Unified Model, which is developed by the UK Met Office and numerous collaborators for weather and climate forecasts.

Should any of this text go in the package doctoring?

File: var/spack/repos/builtin/packages/py-scitools-iris/package.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+class PyScitoolsIris(PythonPackage):
+    """A powerful, format-agnostic, community-driven Python library for
+    analysing and visualising Earth science data"""
+
+    homepage = "http://scitools.org.uk"
+    url      = "https://pypi.io/packages/source/s/scitools-iris/scitools-iris-2.2.0.tar.gz"
+
+    version('2.2.0', sha256='1bf8853f5d7a210f711636d32a52ff62b84a56330fe159720ef56f36f3804ade')
+
+    patch('remove_site_cfg.patch', when='@2.2.0')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-scitools-iris/package.py]: Will this patch be needed on other versions?  Can you explain (in comments) briefly why it's needed at all?

File: var/spack/repos/builtin/packages/py-scitools-iris/package.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-six', type=('build', 'run'))
+    depends_on('py-pyke', type=('build', 'run'))
+    depends_on('py-cartopy', type=('build', 'run'))
+    depends_on('proj@4:4.99')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-scitools-iris/package.py]: Does this really not also depend on `py-proj` (or maybe it's called `py-pyproj`)?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-scitools-iris/package.py]: Based on this line in `requirements/core.txt`:
```python
#conda: proj4<5
```
I'm going to assume that it actually needs `proj`, not `py-pyproj`.

File: var/spack/repos/builtin/packages/py-scitools-iris/package.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    depends_on('py-netcdf4', type=('build', 'run'))
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('py-scipy', type=('build', 'run'))
+
+    depends_on('py-mo-pack', type=('build', 'run'))
+    depends_on('py-nc-time-axis', type=('build', 'run'))
+    depends_on('py-pandas', type=('build', 'run'))
+    depends_on('py-stratify', type=('build', 'run'))
+    depends_on('py-pyugrid', type=('build', 'run'))
+    depends_on('graphviz')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-scitools-iris/package.py]: It looks like this last section of dependencies is optional. Maybe we should add a variant for this?

**Comment by alalazo** [comment]
Closing as stale. Feel free to reopen if you want to continue working on it.

**Pull request closed without merging**